src/spaniel-observer.ts

```diff
@@ -155,20 +163,23 @@ export class SpanielObserver implements SpanielObserverInterface {
   }
   private handleRecordExiting(record: SpanielRecord) {
     const time = Date.now();
+    const perfTime = performance.now();
     record.thresholdStates.forEach((state: SpanielThresholdState) => {
       const boundingClientRect = record.lastSeenEntry && record.lastSeenEntry.boundingClientRect;
       this.handleThresholdExiting(
         {
           intersectionRatio: -1,
           isIntersecting: false,
+          unixTime: time,
           time,
+          highResTime: perfTime,
           payload: record.payload,
           label: state.threshold.label,
           entering: false,
           rootBounds: emptyRect,
           boundingClientRect: boundingClientRect || emptyRect,
           intersectionRect: emptyRect,
-          duration: time - state.lastVisible,
+          duration: state.lastVisible.highResTime - perfTime,
```

Comment:
`perfTime - state.lastVisible.highResTime`? I guess test pass because we calculate `opts.visibleTime = entry.unixTime - entry.duration;` later.

Reply:
oh ya. I'll add a test for the cases where this duration doesn't get overwritten

Reply:
ok looks like this always gets overwritten anyways. really need to re-write this code.